pkg/controller/grafana/grafana_controller.go

```diff
@@ -203,11 +203,14 @@ func (r *ReconcileGrafana) manageError(cr *grafanav1alpha1.Grafana, issue error)
 // Try to find a suitable url to grafana
 func (r *ReconcileGrafana) getGrafanaAdminUrl(cr *grafanav1alpha1.Grafana, state *common.ClusterState) (string, error) {
 	// If preferService is true, we skip the routes and try to access grafana
-	// by using the serivce.
+	// by using the service.
 	preferService := false
 	if cr.Spec.Client != nil {
 		preferService = cr.Spec.Client.PreferService
 	}
+	if preferService != false {
+		cr.Spec.Ingress = nil
```

Comment:
Is this required? There could be use cases where you want the operator to use the service but still create an ingress/route (for example if the route is used in api management like 3scale but operator requests should bypass it).

Reply:
Not required, I'll remove this in the next commit